lxqt-panel 2.2.2 fails to build with C++20

<!--- BEFORE FILLING OUT THIS REPORT FORM:                                 --->
<!--- Dear users of stable and LTS (Long Term Service) distributions:      --->
<!--- Please do NOT file bugs against outdated versions of lxqt-panel,       --->
<!--- which such distributions likely use; instead, use your   --->
<!--- distribution's bugtracker.                                           --->

<!--- Provide a general summary of the issue in the title above. You       --->
<!--- should not delete relevant sections and/or questions in your report  --->

##### Expected Behavior
It should compile with C++20.

##### Current Behavior
Build fails with the following compiler error.

```
plugin-statusnotifier/sniasync.h:90:99: error: no type named 'argument_type' in 'std::function<void (ToolTip)>'
   90 |                     finished(qdbus_cast<typename std::function<typename call_signature<F>::type>::argument_type>(reply.value()));
      |                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
plugin-statusnotifier/sniasync.h:85:17: note: while substituting into a lambda expression here
   85 |                 [this, finished, name] (QDBusPendingCallWatcher * call)
      |                 ^
plugin-statusnotifier/statusnotifierbutton.cpp:281:16: note: in instantiation of function template specialization 'SniAsync::propertyGetAsync<(lambda at plugin-statusnotifier/statusnotifierbutton.cpp:281:59)>' requested here
  281 |     interface->propertyGetAsync(QLatin1String("ToolTip"), [this] (ToolTip tooltip) {
      |                ^
```

##### Possible Solution

╮(￣▽￣"")╭ 

##### Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to  --->
<!--- reproduce this bug. Include code to reproduce, if relevant           --->
1. Build with cmake.

##### Context
<!--- How has this issue affected you? What are you trying to accomplish?  --->
<!--- Providing context helps us come up with a solution that is most      --->
<!--- useful in the real world                                             --->

##### System Information
<!--- Include as many relevant details about the system you experienced    --->
<!--- the bug in                                                           --->
* LXQt Version: 2.2.0
* Distribution & Version: Termux
* Qt Version: 6.9.1
* liblxqt Version: 2.2.0

Here it builds fine but I see a lot  of warnings ` warning: array subscript`
```
   inlined from ‘ToolTip StatusNotifierItemAdaptor::toolTip() const’ at /tmp/makepkg/lxqt-panel-git/src/build/plugin-statusnotifier/statusnotifieritemadaptor.cpp:123:66:
/usr/include/c++/15.1.1/bits/move.h:190:11: warning: array subscript 6 is outside array bounds of ‘QVariant [1]’ [-Warray-bounds=]
  190 |       _Tp __old_val = std::move(__obj);
```
@tsujan ?

--- a/plugin-statusnotifier/sniasync.h
+++ b/plugin-statusnotifier/sniasync.h
@@ -28,15 +28,14 @@
 #if !defined(SNIASYNC_H)
 #define SNIASYNC_H
 
-#include <functional>
 #include "statusnotifieriteminterface.h"
 
 template<typename>
-struct remove_class_type { using type = void; }; // bluff
+struct remove_class_type { using type = void; using arg_type = void; }; // bluff
 template<typename C, typename R, typename... ArgTypes>
-struct remove_class_type<R (C::*)(ArgTypes...)> { using type = R(ArgTypes...); };
+struct remove_class_type<R (C::*)(ArgTypes...)> { using type = R(ArgTypes...); using arg_type = std::tuple_element_t<0, std::tuple<ArgTypes...>>; };
 template<typename C, typename R, typename... ArgTypes>
-struct remove_class_type<R (C::*)(ArgTypes...) const> { using type = R(ArgTypes...); };
+struct remove_class_type<R (C::*)(ArgTypes...) const> { using type = R(ArgTypes...); using arg_type = std::tuple_element_t<0, std::tuple<ArgTypes...>>; };
 
 template <typename L>
 class call_sig_helper
@@ -51,13 +50,13 @@ public:
 template <typename L>
 struct call_signature : public remove_class_type<typename call_sig_helper<L>::type> {};
 template <typename R, typename... ArgTypes>
-struct call_signature<R (ArgTypes...)> { using type = R (ArgTypes...); };
+struct call_signature<R (ArgTypes...)> { using type = R (ArgTypes...); using arg_type = std::tuple_element_t<0, std::tuple<ArgTypes...>>; };
 template <typename R, typename... ArgTypes>
-struct call_signature<R (*)(ArgTypes...)> { using type = R (ArgTypes...); };
+struct call_signature<R (*)(ArgTypes...)> { using type = R (ArgTypes...); using arg_type = std::tuple_element_t<0, std::tuple<ArgTypes...>>; };
 template <typename C, typename R, typename... ArgTypes>
-struct call_signature<R (C::*)(ArgTypes...)> { using type = R (ArgTypes...); };
+struct call_signature<R (C::*)(ArgTypes...)> { using type = R (ArgTypes...); using arg_type = std::tuple_element_t<0, std::tuple<ArgTypes...>>; };
 template<typename C, typename R, typename... ArgTypes>
-struct call_signature<R (C::*)(ArgTypes...) const> { using type = R(ArgTypes...); };
+struct call_signature<R (C::*)(ArgTypes...) const> { using type = R(ArgTypes...); using arg_type = std::tuple_element_t<0, std::tuple<ArgTypes...>>; };
 
 template <typename> struct is_valid_signature : public std::false_type {};
 template <typename Arg>
@@ -87,7 +86,7 @@ public:
                     QDBusPendingReply<QVariant> reply = *call;
                     if (reply.isError() && ignored_errors.cend() == std::find(ignored_errors.cbegin(), ignored_errors.cend(), reply.error().name()))
                         qDebug().noquote().nospace() << "Error on DBus request(" << mSni.service() << ',' << mSni.path() << ',' << name << "): " << reply.error();
-                    finished(qdbus_cast<typename std::function<typename call_signature<F>::type>::argument_type>(reply.value()));
+                    finished(qdbus_cast<typename call_signature<F>::arg_type>(reply.value()));
                     call->deleteLater();
                 }
         );

